### PR TITLE
Reset serializedLayers for lazy serialization in setPaintProperty

### LINKED ISSUE #2651: Style serialization will miss changes to paint properties that don't require re-layout
When @zhangyiatmicrosoft added lazy serialization in #2306, `_updateLayer` was modified to clear the in-memory serialization whenever a layer changed:

https://github.com/maplibre/maplibre-gl-js/blob/d9f276f9fa4960a91ebb502da7559f147f7a58dd/src/style/style.ts#L1195-L1197

The problem is that in `setPaintProperty`, we only call `_updateLayer` if the layer requires re-layout:

https://github.com/maplibre/maplibre-gl-js/blob/d9f276f9fa4960a91ebb502da7559f147f7a58dd/src/style/style.ts#L1071-L1074

I ran into this as a problem trying to change a layer's `raster-resampling` property back and forth between `nearest` and `linear` by updating the style JSON (ie using `setStyle`, not `setPaintProperty`). With the first change, from `nearest` to `linear`, the style diff picked up the change and modified the paint property, but it skipped clearing `this._serializedLayers`. Then, trying to change back to `nearest`, it diffed against the originally serialized value, didn't see any change, and didn't do anything.

I think the simple fix here is just to promote `this._serializedLayers = null` to be called either way in `setPaintProperty`, but it'd be nice to have @zhangyiatmicrosoft 's eyes on this to look for other cases where something like this might be necessary (I'm not very familiar with the original lazy-serialization PR and what kind of audit we did to figure out where the state had to be reset).

cc @ibesora

### PR DESCRIPTION
**Pull request by kumiko-haraguchi**
## Launch Checklist

Fixes #2651

Related to:
- #3373
- #2651

The style's `_serializedLayers` doesn't reset when `setPaintProperty` is called in some cases. It retains the old value when the style is serialized by `map.getStyle()`. To resolve this issue, reset `_serializedLayers` in the `setPaintProperty` method.

[Before]
[before.webm](https://github.com/maplibre/maplibre-gl-js/assets/28984604/0b0c7e6c-53d6-4604-84e4-8651e4268d2e)

[After]
[after.webm](https://github.com/maplibre/maplibre-gl-js/assets/28984604/36aeff0d-8534-4144-958d-c0bf86e7d326)

----
I also noticed that the documentation includes private methods [_serializebyids](https://maplibre.org/maplibre-gl-js/docs/API/classes/Style/#_serializebyids), [_serializedalllayers](https://maplibre.org/maplibre-gl-js/docs/API/classes/Style/#_serializedalllayers) for the Style class. I removed the doc comments for these methods, as they may not have been intended to be documented.

----
 - [x] Confirm **your changes do not include backports from Mapbox projects** (unless with compliant license) - if you are not sure about this, please ask!
 - [x] Briefly describe the changes in this PR.
 - [x] Link to related issues.
 - [x] Include before/after visuals or gifs if this PR includes visual changes.
 - [x] Write tests for all new functionality.
 - [x] Add an entry to `CHANGELOG.md` under the `## main` section.

### CODE REVIEW

File: src/ui/map.test.ts [hunk 1]
```diff
@@ -1818,6 +1818,33 @@ describe('Map', () => {
             });
         });
 
+        test('#3373 paint property should be synchronized with an update', done => {
```
Comment (HarelM) [inline src/ui/map.test.ts:1821]: Can we avoid using `done` in tests? It doesn't give a good error message when it fails and using `once` and `async/await` is more readable.

File: src/style/style.ts [hunk 2]
```diff
@@ -470,11 +470,6 @@ export class Style extends Evented {
         return true;
     }
 
-    /**
```
Comment (HarelM) [inline src/style/style.ts:473]: Why were these comments removed?

Comment (kumiko-haraguchi) [reply src/style/style.ts:473]: I noticed that the documentation includes private methods [_serializebyids](https://maplibre.org/maplibre-gl-js/docs/API/classes/Style/#_serializebyids), [_serializedalllayers](https://maplibre.org/maplibre-gl-js/docs/API/classes/Style/#_serializedalllayers) for the Style class. I removed these doc comments for these methods, as they may not have been intended to be documented. I'll put them back if these are intended!

Style Doc

Comment (HarelM) [reply src/style/style.ts:473]: Ahh, in that case I suggest to use `@hidden` instead, these comments can be meaningful when trying to understand the code.

Comment (kumiko-haraguchi) [reply src/style/style.ts:473]: @HarelM Sure, thanks for the suggestion!

**Review by HarelM — APPROVED** [review]

**Pull request merged**